app/src/db/models/base.py

```diff
@@ -26,10 +25,31 @@
 )
 
 
-@as_declarative(metadata=metadata)
-class Base:
+class Base(DeclarativeBase):
+    # Attach the metadata to the Base class so all tables automatically get added to the metadata
+    metadata = metadata
+
+    # Override the default type that SQLAlchemy will map python types to.
+    # This is used if you simply define a column like:
+    #
+    #   my_column: Mapped[str]
+    #
+    # If you provide a mapped_column attribute you can override these values
+    #
+    # See: https://docs.sqlalchemy.org/en/20/orm/declarative_tables.html#mapped-column-derives-the-datatype-and-nullability-from-the-mapped-annotation
+    #      for the default mappings
+    #
+    # See: https://docs.sqlalchemy.org/en/20/orm/declarative_tables.html#orm-declarative-mapped-column-type-map
+    #      for details on setting up this configuration.
+    type_annotation_map = {
+        # Explicitly use the Text column type for strings
+        str: Text,
+        # Always include a timezone for datetimes
+        datetime: TIMESTAMP(timezone=True),
```

Comment:
nonblocking question: any idea why sqlalchemy makes `TIMESTAMP` all caps?

Reply:
I'm not sure, in some cases they have a normal-cased one like with Text, but not in this case.